### DISCUSSION TITLE
"Transform" Centroids into GeodataFrames

### DISCUSSION THREAD

**Discussion by chahank** [CLIMADA-project/climada_python#710] [Ideas]
The centroids objects are essentially a manually created `GeoDataframe`. The code could be simplified by making an attribute `.gdf` to hold the data. The only attribute that breaks this currently is the `meta` to describe rasters. However, since all of CLIMADA internally works with vector data and does not support rasters, we could drop this and only use rasters in I/O operations.

**Comment by peanutfun** [comment]
Excellent idea! This would probably simplify a lot of the internals. The way I see it, all attributes except `meta` can be stored as columns in the `GeoDataFrame`. Some backwards-compatibility (for both the internals and external access) can be provided through accessors like
```python
@property
def elevation(self):
    return self.gdf["elevation"]
```
